synapse/storage/databases/main/receipts.py

```diff
@@ -764,6 +765,10 @@ async def insert_receipt(
                 linearized_event_id,
                 data,
                 stream_id=stream_id,
+                # Read committed is actually beneficial here because we check for a receipt with
+                # greater stream order, and checking the very latest data at select time is better
+                # than the data at transaction start time.
+                isolation_level=IsolationLevel.READ_COMMITTED,
```

Comment:
I'm a bit confused as to *why* anything is changing during this transaction TBH. Are we ending up inserting multiple receipts for the same user/room at the same time?